zlib-rs/src/inflate.rs

```diff
@@ -1835,3 +2159,18 @@ fn init_window<'a>(
 
     Ok(window)
 }
+
+pub fn get_header<'a>(stream: &'a mut InflateStream<'a>, head: &'a mut GzipHeader) -> ReturnCode {
+
+    // TODO Why does this always fail?
+    //if (stream.state.wrap & 2) == 0 {
+    //    return ReturnCode::StreamError;
+    //}
```

Comment:
guess: you need a state that is initialized with a memory level in the range `16..32`